src/sql/src/statements/transform/expand_interval.rs

```diff
@@ -65,8 +65,8 @@ impl TransformRule for ExpandIntervalTransformRule {
     /// it's AST has `left` part of type `Value::SingleQuotedString` which needs to be handled specifically.
     /// To handle the `right` part which is `Interval` no extra steps are needed.
     fn visit_expr(&self, expr: &mut Expr) -> ControlFlow<()> {
-        if let Expr::Interval(interval) = expr {
-            match *interval.value.clone() {
+        match expr {
+            Expr::Interval(interval) => match *interval.value.clone() {
```

Comment:
We can avoid the clone if it's not the expected pattern.

Try to match on the references: `&*interval.value` and `&**left `,

```rust
            Expr::Interval(interval) => match &*interval.value {
                Expr::Value(Value::SingleQuotedString(value))
                | Expr::Value(Value::DoubleQuotedString(value)) => {
                    if let Some(data) = expand_interval_name(value) {
                        *expr = create_interval_with_expanded_name(
                            interval,
                            single_quoted_string_expr(data),
                        );
                    }
                }
                Expr::BinaryOp { left, op, right } => match &**left {
                    Expr::Value(Value::SingleQuotedString(value))
                    | Expr::Value(Value::DoubleQuotedString(value)) => {
                        if let Some(data) = expand_interval_name(value) {
                            let new_value = Box::new(Expr::BinaryOp {
                                left: single_quoted_string_expr(data),
                                op: op.clone(),
                                right: right.clone(),
                            });
                            *expr = create_interval_with_expanded_name(interval, new_value);
                        }
                    }
                    _ => {}
                },
                _ => {}
            },

```

Reply:
@killme2008 Thanks for pointing that out 🙏! 
That's something where I definitely need to improve!

Reply:
You are welcome!